src/utils/localDevelopmentUtils.ts

```diff
@@ -0,0 +1,152 @@
+import { HttpOperationResponse } from '@azure/ms-rest-js';
+import * as extract from 'extract-zip';
+import { URL } from 'url';
+import * as vscode from 'vscode';
+import { callWithTelemetryAndErrorHandling, IActionContext } from 'vscode-azureextensionui';
+import { ProjectLanguage } from '../constants';
+import { AzureAccount } from '../debug/AzureAccountExtension.api';
+import { GlobalStates } from '../extension';
+import { ext } from '../extensionVariables';
+import { localize } from '../localize';
+import { getNameFromId } from './azure';
+import { requestUtils } from './requestUtils';
+
+interface ILocalDevelopmentRequiredInputs {
+    resourceId: string | null;
+    defaultHostName: string | null;
+    devContainerName: string | null;
+    language: string | null;
+    downloadAppContent: string | null;
+}
+
+export namespace localDevelopmentUtils {
```

Comment:
"localDevelopment" could describe over half the features in our extension and "utils" is a last resort folder for small utility functions that can't be placed anywhere better because they're used by multiple features. I would appreciate if you took some time familiarizing yourself with our file structure and made an attempt at a better location/name. If you're still struggling to find a place after you've addressed the other comments, I'll come up with a suggestion at that time

Reply:
I have moved this around. If there is still a concern, it would be better to discuss it over a call to expedite this process.